### DISCUSSION TITLE
How to get output with safeParse without checking for success

### DISCUSSION THREAD

**Discussion by ericmartinezr** [open-circle/valibot#282] [Q&A] [answered]
Hello!

My first question here.

I'm trying to get the output out of safeParse without checking directly for success but after checking for **not success**. Let me explain myself.

```ts
const data = ...;
const someSchema = schema(...);
const result = safeParse(someSchema, data);
if (!result.success) {
   // Throw some error
}
// This should probably work, we already went through "not success"
const MyObject = result.output
```
The above snipet shows that **output** doesn't exist if we're not checking directly for success. I have a relatively huge amount of code that should happen when it's parsed successfuly but I don't want to enclose all of it inside a `if (parse.success) {}`, by checking that's it's **not success** should be enough imho.

Am I doing something wrong?

Thanks in advance!

**Comment by fabian-hiller** [comment]
Your approach works fine for me. Did you set `strict` to `true` in your `tsconfig.json`?

**Reply by ericmartinezr** [reply]
Yes, I did set `strict` to true because valibot requires/recommends it.

**Reply by fabian-hiller** [reply]
![Screenshot 2023-12-04 at 09 34 23](https://github.com/fabian-hiller/valibot/assets/35291865/2c699bf3-27a6-4d0f-b636-cbf9a33ce587)

**Reply by fabian-hiller** [reply]
What version are you using? Have you upgraded to v0.22.0?

**Reply by ericmartinezr** [reply]
I'm using v0.21.0 but I noticed now thanks to your example that the version is not the problem.

I'm not sure if this is normal but this is what I'm doing 

![image](https://github.com/fabian-hiller/valibot/assets/13039652/c832b7dd-befb-4da5-8569-9d284787e7f7)


but if I throw inside the if it works like just in your example

![image](https://github.com/fabian-hiller/valibot/assets/13039652/c15aec94-b492-4780-959b-40335d64a34a)

Not sure if that's intentional or not, but I will see to throw directly inside the if instead of using a function, I would prefer the function though.

Thanks for your help @fabian-hiller and sorry for the noise.

**Reply by fabian-hiller** [reply, marked as answer]
I think TypeScript needs to see the `throw` statement. So you could change your code like this:

```ts
private _rethrow() {
  return new Error('Invalid');
}

async guarder() {
  // ...
  if (!parseResult.success) {
    throw this._rethrow()
  }
  const data = parseResult.output;
}
```

**Reply by ericmartinezr** [reply]
Thanks a lot! That worked!
